File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int size = nums.size();
        vector<int> output(size, 1);
        int prefix = 1;
        for (int i = 0; i < size; i++)
        {
            output[i] = prefix;
            prefix *= nums[i];
        }

        int postfix = 1;
        for (int i = size - 1; i >= 0; i--)
        {
            output[i] *= postfix;
            postfix *= nums[i];
        }

        return output;
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        output = []
        size = len(nums)
        nums.sort()
        def helper(start = 0, curr = []):
            nonlocal output
            output.append(curr[:])

            for idx in range(start, size): # start = 1, idx = 0
                if idx > start and nums[idx] == nums[idx - 1]:
                    continue
                curr.append(nums[idx])
                helper(idx + 1, curr)
                curr.pop()
        helper()

        return output